### PR TITLE
`azurerm_app_service_environment` - add support for ASEV3 Preview

### LINKED ISSUE #9420: app_service_environment v3
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

* terraform 0.12.24
* azurerm 2.37

### Affected Resource(s)

* `azurerm_app_service_environment`

### Terraform Configuration Files

```hcl
resource "azurerm_app_service_environment" "default" {
  name                         = "ase-v3-example"
  subnet_id                    = azurerm_subnet.default.id
  pricing_tier                 = "I1v2"
  front_end_scale_factor       = 10
  internal_load_balancing_mode = "Web, Publishing"
}
```

### Expected Behaviour

Creates an ASE v3 instance.

### Actual Behaviour

`expected pricing_tier to be one of [I1 I2 I3], got I1v2`

### Steps to Reproduce

1. `terraform apply`

### References

ASE v3 is in public preview as of Nov 1: https://docs.microsoft.com/en-us/azure/app-service/environment/creation

* #0000

### PR DESCRIPTION
**Pull request by jackofallops**
Resolves #9420

Needs #10303 to be merged first, in draft until then.

### CODE REVIEW

File: website/docs/r/app_service_environment.html.markdown [hunk 1]
```diff
@@ -62,7 +62,9 @@ resource "azurerm_app_service_environment" "example" {
 
 * `subnet_id` - (Required) The ID of the Subnet which the App Service Environment should be connected to. Changing this forces a new resource to be created.
 
-~> **NOTE** a /24 or larger CIDR is required. Once associated with an ASE this size cannot be changed.
+~> **NOTE:** a /24 or larger CIDR is required. Once associated with an ASE this size cannot be changed.
```
Comment (katbyte) [inline website/docs/r/app_service_environment.html.markdown:65]:
```suggestion
~> **NOTE:** a `/24` or larger CIDR is required. Once associated with an ASE this size cannot be changed.
```

**Review by katbyte — APPROVED** [review]
Aside from one minor comment this LGTM 👍

File: website/docs/r/app_service_environment.html.markdown [hunk 2]
```diff
@@ -78,6 +80,10 @@ resource "azurerm_app_service_environment" "example" {
 
 * `resource_group_name` - (Optional) The name of the Resource Group where the App Service Environment exists. Defaults to the Resource Group of the Subnet (specified by `subnet_id`).
 
+* `version` - (Optional) The Version of the Application Service Environment. Possible values are `ASEV2` and `ASEV3`. Defaults to `ASEV2`.
```
Comment (tombuildsstuff) [inline website/docs/r/app_service_environment.html.markdown:83]: thinking about the deprecation/removal path for v2 down the line here: would this make more sense to split out into a `_v3` resource?

Comment (jackofallops) [reply website/docs/r/app_service_environment.html.markdown:83]: The preview currently "fits" within the existing resource, and my original thought was to be able to deprecate fields over time as Azure promotes V3 to the default option.  That said, there's enough difference in implementation to warrant a separate resource. I'll pick up with you offline to discuss before assigning this PR to a release.

**Comment by jackofallops** [comment]
Closing in favour of 11174

**Pull request closed without merging**